src/test/regress/sql/rpt.sql

```diff
@@ -562,6 +562,51 @@ explain (costs off, verbose) insert into t2 (a, b) select i, random() from t;
 explain (costs off, verbose) select * from t1 where a in (select f(i) from t where i=a and f(i) > 0);
 -- ensure we do not break broadcast motion
 explain (costs off, verbose) select * from t1 where 1 <= ALL (select i from t group by i having random() > 0);
+reset enable_bitmapscan;
+reset enable_seqscan;
+set gp_cte_sharing = on;
+-- ensure that the volatile function is executed on one segment if it is in the CTE target list
+explain (costs off, verbose) with cte as (
+    select a * random() as a from generate_series(1, 5) a
+)
+select * from cte join (select * from t1 join cte using(a))b using(a);
```

Comment:
I'm ready to approve. But firstly, check all the tests on missing spaces as here `))b`.

Reply:
And write queries with cte in the same style. First you write `with cte as ( \n`,  then you write ` with cte as (select a, a * random() from generate_series(1, 5) a)`